recipes/faker-cxx/all/conanfile.py

```diff
@@ -77,10 +93,10 @@ def source(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["USE_SYSTEM_DEPENDENCIES"] = True
-        tc.variables["BUILD_TESTING"] = False
-        tc.variables["WARNINGS_AS_ERRORS"] = False
-        tc.variables["WITH_STD_FORMAT"] = self.options.with_std_format
+        tc.cache_variables["USE_SYSTEM_DEPENDENCIES"] = True
+        tc.cache_variables["BUILD_TESTING"] = False
+        tc.cache_variables["WARNINGS_AS_ERRORS"] = False
+        tc.cache_variables["USE_STD_FORMAT"] = self._with_std_format
```

Comment:
`WITH_STD_FORMAT` is not a variable in any of the supported versions, moved this to use the correct `USE_STD_FORMAT`. Note that in old versions, when `USE_SYSTEM_DEPENDENCIES` was true, fmt was always looked for regardless of the value of the flag, which saved the recipe from failing